amethyst_network/src/error.rs

```diff
@@ -0,0 +1,85 @@
+//! Module containing error handling logic.
+
+use std::{
+    fmt::{self, Display, Formatter},
+    io,
+    sync::mpsc,
+};
+
+use crate::server::ServerSocketEvent;
+
+/// The `amethyst_network` result type.
+pub type Error<T> = Result<T, ErrorKind>;
+
+/// Wrapper for all errors who could occur in `amethyst_network`.
+#[derive(Debug)]
+pub enum ErrorKind {
```

Comment:
Make use of the [`__NonExhaustive` pattern](https://github.com/rust-lang/rfcs/blob/master/text/2008-non-exhaustive.md#how-we-do-this-today) so we guard against compatibility issues in case someone pattern matches over the type.